moveit_core/collision_detection_fcl/include/moveit/collision_detection_fcl/collision_detector_fcl_plugin_loader.h

```diff
@@ -0,0 +1,19 @@
+/*
```

Comment:
This file should be available by cherry picking in https://github.com/ros-planning/moveit/commit/d8a9337d3a2ce0b72b661cb15dc394e72f93e2f5. I don't think it is a good idea to copy-paste in this file without the rest of the [PR](https://github.com/ros-planning/moveit/pull/1370/files)?

@BryceStevenWilley can you comment on this?

Reply:
Agreed, cherry picking that PR should be easier. This PR is missing the [plugin XML](https://github.com/ros-planning/moveit/commit/d8a9337d3a2ce0b72b661cb15dc394e72f93e2f5#diff-11ca90ef7e928f981c48989bafa3c3fa), and the [modification](https://github.com/ros-planning/moveit/commit/d8a9337d3a2ce0b72b661cb15dc394e72f93e2f5#diff-32447bddfd3c0d47bc0c0dba873ea1b6) to the package.xml. FCL will still load by default, but things might break when you try to manually load the plugin. However, I can't speak to how pluginlib works in ROS2/if any of these files need to be modified.

Reply:
@vmayoral As your team tackles pluginlib in ros2, it may be helpful to have examples to draw from. RQT2 and RVIZ2 both use pluginlib. 

When I was porting the CPP plugin provider for RQT2 this README in the rviz2 repo was really useful: https://github.com/ros2/rviz/blob/ros2/docs/plugin_development.md#plugin-basics

rqt_gui_cpp in the rqt repo uses the new pluginlib for ros2. It may be helpful to look at the diff between [crystal-devel and kinetic-devel](https://github.com/ros-visualization/rqt_image_view/compare/master...crystal-devel) and reference the changes to rqt_gui_cpp.

rqt_image_view is another good example of a plugin that gets registered and loaded by pluginlib:
[branch](https://github.com/ros-visualization/rqt_image_view/tree/crystal-devel)
([Diff between master and crystal-devel](https://github.com/ros-visualization/rqt_image_view/compare/master...crystal-devel))

Reply:
Thanks @mlautman having look now. Experiencing some issues with rviz2 though https://github.com/ros2/rviz/issues/385 in OS X.